sdk/src/main/java/io/dapr/client/domain/BulkPubSubMessage.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2023 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io.dapr.client.domain;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents a bulk of messages received from the message bus.
+ */
+public final class BulkPubSubMessage<T> {
+  private final List<BulkPubSubMessageEntry<T>> entries;
+  private final String topic;
+  private final Map<String, String> metadata;
+
+  /**
+   * Instantiate a BulkPubSubMessage.
+   * @param entries mapping from bulk pubSub message entry ID to a status.
+   * @param topic pubSub topic.
+   * @param metadata metadata for the bulk message.
+   */
+  @JsonCreator
```

Comment:
Sorry for this but we should not have dependency on Jackson in our domain objects. Now, it seems that this is only required at the sdk-springboot. So, I would create an inherited class from his one with the same name but in a java package called `io.dapr.client.domain.jackson` in sdk-springboot project. In the docs, you make it clear that the jackson package should be used instead.

Another options (which might be better) is to make all these classes specific to BulkSubscribe and remove any "PubSub" from the name in favor of "Subscribe", moving all of it to the sdk-springboot project in the `io.dapr.client.domain.jackson` package, avoiding any duplication of code.

Reply:
FYI we do depend on Jackson already in domain objects, example -

https://github.com/dapr/java-sdk/blob/b83661d1094361a6c41a9681d903a4d064687106/sdk/src/main/java/io/dapr/client/domain/SubscribeConfigurationResponse.java#L43-L49

Reply:
As discussed offline, skipping this change for now. Only renamed `PubSub` => `Subscribe`.